File: app/src/main/java/id/kotlin/sample/movie/data/local/Movie.java
package id.kotlin.sample.movie.data.local;

import android.os.Parcel;
import android.os.Parcelable;

public class Movie implements Parcelable {

    public String title;
    public String desc;
    public String date;
    public String image;
    public double vote;

    public static final Creator<Movie> CREATOR = new Creator<Movie>() {
        @Override
        public Movie createFromParcel(Parcel source) {
            return new Movie(source);
        }

        @Override
        public Movie[] newArray(int size) {
            return new Movie[size];
        }
    };

    public Movie() {
    }

    protected Movie(Parcel in) {
        this.title = in.readString();
        this.desc = in.readString();
        this.date = in.readString();
        this.image = in.readString();
        this.vote = in.readDouble();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.title);
        dest.writeString(this.desc);
        dest.writeString(this.date);
        dest.writeString(this.image);
        dest.writeDouble(this.vote);
    }
}

File: app/src/main/java/id/kotlin/sample/movie/data/local/Movie.kt
package id.kotlin.sample.movie.data.local

import android.os.Parcel
import android.os.Parcelable

data class Movie(@JvmField var title: String,
                 @JvmField var desc: String,
                 @JvmField var date: String,
                 @JvmField var image: String,
                 @JvmField var vote: Double) : Parcelable {

    constructor() : this(
            title = "",
            desc = "",
            date = "",
            image = "",
            vote = 0.0
    )

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Movie> = object : Parcelable.Creator<Movie> {
            override fun createFromParcel(source: Parcel): Movie = Movie(source)
            override fun newArray(size: Int): Array<Movie?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readString(),
            source.readString(),
            source.readString(),
            source.readString(),
            source.readDouble()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(title)
        dest.writeString(desc)
        dest.writeString(date)
        dest.writeString(image)
        dest.writeDouble(vote)
    }
}